pkg/controller/deployment/deployment_controller.go

```diff
@@ -611,7 +627,13 @@ func (dc *DeploymentController) syncDeploymentStatus(allRSs []*extensions.Replic
 		return err
 	}
 	if d.Generation > d.Status.ObservedGeneration || d.Status.Replicas != totalActualReplicas || d.Status.UpdatedReplicas != updatedReplicas || d.Status.AvailableReplicas != availableReplicas {
-		return dc.updateDeploymentStatus(allRSs, newRS, d)
+		var cond *extensions.DeploymentCondition
+		if deploymentutil.IsProgressing(*d) {
+			// Mark the deployment as complete so that the lack of progress won't result in
+			// falsely marking it as failed.
+			cond = deploymentutil.NewCondition(extensions.DeploymentComplete, api.ConditionTrue, "Completed", "Deployment progress is complete")
+		}
+		return dc.updateDeploymentStatus(allRSs, newRS, d, cond)
```

Comment:
Hm, we need a way to separate between scaling events and rollouts otherwise the `d.Generation == d.Status.ObservedGeneration && d.Status.UpdatedReplicas == d.Spec.Replicas` won't be enough. I think PDS should have effect only on rollouts and not during scaling events. Thoughts?